src/analysis/validation.jl

```diff
@@ -2,102 +2,238 @@
 """
 Perform EPANET hydraulic simulation (via WNTR) and compute timeseries of flows and heads
 """
-function validate(data::Dict{String,Any}, solution::Dict{String,Any},
-                     inpfilepath::String, wn, wnres)
 
-    # for each node, compute the head values
-    # for each link, compute the flow rates and head gains
-    
-    wslinks = wnres.link
-    wsnodes = wnres.node
-
-    # ------------------------------------------#
-    #                   node                    #
-    # ------------------------------------------#
-    wntr_head = wsnodes["head"]
-
-    node_head_wntr = Dict{String,Any}()
-    node_head_wm = Dict{String,Any}()
-
-    for (key, node) in solution["1"]["node"] 
-        node_name = data["node"][key]["source_id"][2]
-        node_head_wntr[node_name] = Array{Float64,1}(undef,length(solution))
-        node_head_wm[node_name] = Array{Float64,1}(undef,length(solution))
-    end
+using DataFrames
 
-    for tx in 1:length(keys(solution))
-        for (key, node) in solution[string(tx)]["node"] 
-            node_name = data["node"][key]["source_id"][2]
-            if haskey(wntr_head, node_name)
-                node_head_wntr[node_name][tx] = wntr_head[node_name].values[tx]
-                node_head_wm[node_name][tx] = node["h"]
-            end
-        end
-    end
+# compare nodes
+function get_node_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation, node_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
```

Comment:
This should be called `num_steps` or something similar instead of `duration`. Duration should refer to something measured in seconds.

Reply:
Looks like this has been fixed.